viz_scripts/generic_timeseries.ipynb

```diff
@@ -136,6 +147,30 @@
    },
    "outputs": [],
    "source": [
+    "def compute_daily_metrics(data, weight_unit, distance_unit):\n",
+    "    # Sum daily distance traveled for each mode\n",
+    "    mode_distance = data.groupby(['user_id','date_time','Mode_confirm'], as_index=False)[[f'{distance_unit}']].sum()\n",
+    "    mode_distance.rename(columns={'sum':f'{distance_unit}'}, inplace=True)\n",
+    "    mode_distance[f'{distance_unit}'] = mode_distance[f'{distance_unit}'].fillna(0)\n",
+    "    \n",
+    "    # Sum daily emissions for each user\n",
+    "    emissions = data.groupby(['user_id','date_time'], as_index=False)[[f'Mode_confirm_{weight_unit}_CO2', f'{distance_unit}']].sum()\n",
+    "    emissions[f'Mode_confirm_{weight_unit}_CO2'] = emissions[f'Mode_confirm_{weight_unit}_CO2'].fillna(0)\n",
+    "    emissions[f'{distance_unit}'] = emissions[f'Mode_confirm_{weight_unit}_CO2'].fillna(0)\n",
+    "    \n",
+    "    # Sum daily energy for each user\n",
+    "    energy = data.groupby(['user_id','date_time'], as_index=False)[['Mode_confirm_EI(kWH)', f'{distance_unit}']].sum()\n",
+    "    energy['Mode_confirm_EI(kWH)'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
+    "    energy[f'{distance_unit}'] = energy['Mode_confirm_EI(kWH)'].fillna(0)\n",
+    "    \n",
+    "    # Add 7-day rolling avg smoothing to better see trends\n",
+    "    mode_counts['trip_count_smooth'] = mode_counts.groupby(['user_id','Mode_confirm'])['trip_count'].apply(lambda x: x.rolling(7,1).mean())\n",
+    "    mode_distance[f'{distance_unit}smooth'] = mode_distance.groupby(['user_id','Mode_confirm'])[f'{distance_unit}'].apply(lambda x: x.rolling(7,1).mean())\n",
```

Comment:
I don't think this is actually correct. How did you test this? I don't see how it can work without the underscore.

```suggestion
    "    mode_distance[f'{distance_unit}_smooth'] = mode_distance.groupby(['user_id','Mode_confirm'])[f'{distance_unit}'].apply(lambda x: x.rolling(7,1).mean())\n",
```

Reply:
I have tested the `generic_timeseries` notebook with both `default mapping` and `dynamic config` through running Jupyter notebook automatically as mentioned above.
Interestingly, we don't use `mode_distance` after we have called `mode_distance[f'{distance_unit}_smooth']`. This is an existing code changes. I am not sure if this was written to handle future things, but at the moment, it does not execute anything over `mode_distance`.
Still, updated the above code to use `mode_distance[f'{distance_unit}_smooth']`.

Reply:
We do use it to compute "Emissions per mile per day"

```
try:
    # Emissions per mile per day across all users (travel efficiency)
    # Note that the energy plot will be identical to this one since scale factor is divided out
    emissions['CO2_per_mile'] = emissions['Mode_confirm_lb_CO2'] / emissions['distance_miles_smooth']
    emissions['CO2_per_mile'] = emissions['CO2_per_mile'].fillna(0)
    plot_data = emissions.groupby(['date_time'])['CO2_per_mile'].agg(['mean']).reset_index()
```

Is this not in the new codebase?

Reply:
Ah I see, that uses `emissions['distance_miles_smooth']` and not `mode_distance[f'{distance_unit}_smooth']`

Reply:
It _is_ used in `mode_specific_timeseries`, though

Reply:
And I don't see any related changes in `mode_specific_timeseries`. Ah but that is because `mode_specific_timeseries` does not appear to have any code related to emission calculations.
@iantei I would like to see such investigation and clarification/documentation in the future.